resources/fetchersManager/data.go

```diff
@@ -165,9 +168,18 @@ func (d *Data) fetchProtected(ctx context.Context, k string, metadata fetching.C
 // Stop cleans up Data resources gracefully.
 func (d *Data) stopData(ctx context.Context, grace time.Duration) {
 	d.stopNotice <- grace
+
+	startTime := time.Now()
 	ctx, cancel := context.WithTimeout(ctx, grace)
 	defer cancel()
 	<-ctx.Done()
 	d.fetchers.Stop()
 	close(d.stop)
+
+	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+		d.log.Debug("Context timeout exceeded")
+	}
+	if errors.Is(ctx.Err(), context.Canceled) {
+		d.log.Debug("Context canceled", time.Since(startTime).String())
+	}
```

Comment:
you can test this branch by plugging the periods from the table in the description and looking at the logs to find the relevant line.